ddtrace/tracer/telemetry_test.go

```diff
@@ -0,0 +1,77 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package tracer
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/telemetry"
+)
+
+func TestTelemetryEnabled(t *testing.T) {
+	t.Setenv("DD_TRACE_STARTUP_LOGS", "0")
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	received := make(chan *telemetry.AppStarted, 1)
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path != "/telemetry/proxy/api/v2/apmtelemetry" {
+			return
+		}
+		if r.Header.Get("DD-Telemetry-Request-Type") != string(telemetry.RequestTypeAppStarted) {
+			return
+		}
+		var body telemetry.Body
+		body.Payload = new(telemetry.AppStarted)
+		err := json.NewDecoder(r.Body).Decode(&body)
+		if err != nil {
+			t.Errorf("bad body: %s", err)
+		}
+		select {
+		case received <- body.Payload.(*telemetry.AppStarted):
+		default:
+		}
+	}))
+	defer server.Close()
+
+	Start(
```

Comment:
I would have expected this test to verify behavior of the functions provided by this package, but it actually calls `Start` and `Stop` directly instead of `startTelemetry` and `stopTelemetry`. Is that intentional, or is there a way to unit test those functions instead?

Reply:
Previously the test and `start/stopTelemetry` functions were in `tracer/tracer_test.go`, but I decided to move it into a separate file to have all telemetry related helper functions in one place for now and the future. 

My intent with this specific test is to verify that telemetry info is sent by default on tracer start. We can also directly unit test `start/stopTelemetry` by just passing in a tracer config struct, but that doesn't cover the logic of `tracer start --> telemetry sent` as completely. Do you think this unit test should be moved back to `test_tracer.go`?